default/python/AI/ColonisationAI.py

```diff
@@ -651,13 +660,14 @@ def evaluate_planet(planet_id, mission_type, spec_name, detail=None):
     planet_supply += sum(AIDependencies.building_supply[bld_type].get(int(psize), 0)
                          for psize in [-1, planet.size] for bld_type in bld_types)
     planet_supply += sum(AIDependencies.SUPPLY_MOD_SPECIALS[_special].get(int(psize), 0)
-                         for psize in [-1, planet.size] for _special in
-                         set(planet.specials).union(system.specials).intersection(AIDependencies.SUPPLY_MOD_SPECIALS))
+                         for _special in set(planet.specials).union(system.specials).intersection(AIDependencies.SUPPLY_MOD_SPECIALS)
```

Comment:
You can extract it to the variable for readability
`set(planet.specials).union(system.specials).intersection(AIDependencies.SUPPLY_MOD_SPECIALS)`